Library/Homebrew/brew.sh

```diff
@@ -6,7 +6,10 @@ case "$HOMEBREW_SYSTEM" in
 esac
 
 # Force UTF-8 to avoid encoding issues for users with broken locale settings.
-if [[ "$(locale charmap 2>/dev/null)" != "UTF-8" ]]
+if ! command -v locale >/dev/null
+then
+  export LC_ALL=C
```

Comment:
This isn't correct behaviour on macOS, unfortunately.

Reply:
Under what circumstances does `command -v locale` fail on macOS? I believe that situation is impossible with SIP enabled.

Reply:
It is hence I'd like to avoid the check.

Reply:
I can rewrite this part using `if [[ -n "$HOMEBREW_MACOS" ]]; then ... else ... {complex logic}; fi` -- would that work?

Reply:
Yeh, go for it. I'm still of the belief that checking for `locale` is pointless where we're correctly handling the case when it doesn't exist anyway if we hide the errors.